limitador/src/storage/in_memory.rs

```diff
@@ -1,31 +1,98 @@
 use crate::counter::Counter;
 use crate::limit::{Limit, Namespace};
+use crate::storage::expiring_value::ExpiringValue;
 use crate::storage::{Authorization, CounterStorage, StorageErr};
 use std::collections::hash_map::Entry;
 use std::collections::{HashMap, HashSet};
+use std::hash::{Hash, Hasher};
 use std::sync::RwLock;
-use std::time::Duration;
-use ttl_cache::TtlCache;
+use std::time::{Duration, SystemTime};
+
+#[derive(Eq, Clone)]
+struct CounterKey {
+    set_variables: HashMap<String, String>,
+}
+
+impl CounterKey {
+    fn to_counter(&self, limit: &Limit) -> Counter {
+        Counter::new(limit.clone(), self.set_variables.clone())
+    }
+}
+
+impl From<&Counter> for CounterKey {
+    fn from(counter: &Counter) -> Self {
+        CounterKey {
+            set_variables: counter.set_variables().clone(),
+        }
+    }
+}
+
+impl From<&mut Counter> for CounterKey {
+    fn from(counter: &mut Counter) -> Self {
+        CounterKey {
+            set_variables: counter.set_variables().clone(),
+        }
+    }
+}
+
+impl Hash for CounterKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.set_variables.iter().for_each(|(k, v)| {
+            k.hash(state);
+            v.hash(state);
+        });
+    }
+}
+
+impl PartialEq for CounterKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.set_variables == other.set_variables
+    }
+}
 
 pub struct InMemoryStorage {
-    limits_for_namespace: RwLock<HashMap<Namespace, HashMap<Limit, HashSet<Counter>>>>,
-    counters: RwLock<TtlCache<Counter, i64>>,
+    //TODO: This is a bit ugly, to address in the future
+    #[allow(clippy::type_complexity)]
+    limits_for_namespace:
+        RwLock<HashMap<Namespace, HashMap<Limit, HashMap<CounterKey, ExpiringValue>>>>,
```

Comment:
Do you think you'd eventually make this simpler? Otherwise a type alias might be a good idea already, wdyt?